server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java

```diff
@@ -303,6 +307,9 @@ public Analysis visitQuery(QueryStatement queryStatement, MPPQueryContext contex
       } else {
         Map<Integer, List<Pair<Expression, String>>> outputExpressionMap =
             analyzeSelect(analysis, queryStatement, schemaTree);
+        if (outputExpressionMap.values().stream().allMatch(List::isEmpty)) {
+          return finishQuery(analysis);
+        }
```

Comment:
It's better to check `outputExpressions` below. That way we don't need to traverse the outputExpressionMap twice.